From 81e2a8401115d272d419d58536bc1592e60f4a27 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:11:44 -0400
Subject: [ethernet] i40evf: don't lie to ethtool

Message-id: <1409663591-32531-62-git-send-email-sassmann@redhat.com>
Patchwork-id: 90540
O-Subject: [RHEL7.1 PATCH 061/148] i40evf: don't lie to ethtool
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

Don't tell ethtool that the VF can do 10GbaseT, when it really has no
idea what its link speed is. Set the supported values to 0 instead.
Suggested by Ben Hutchings.

CC: Ben Hutchings <ben@decadent.org.uk>
Change-ID: Iceb0d8af68fe5d8dc13224366979ba701ba89c39
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit 107f301861fd29b3bec9e76278cb87da9990aa11)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40evf_ethtool.c b/drivers/net/ethernet/intel/i40evf/i40evf_ethtool.c
index 3566e73..b0f6270 100644
--- a/drivers/net/ethernet/intel/i40evf/i40evf_ethtool.c
+++ b/drivers/net/ethernet/intel/i40evf/i40evf_ethtool.c
@@ -80,7 +80,7 @@ static int i40evf_get_settings(struct net_device *netdev,
  /* In the future the VF will be able to query the PF for
   * some information - for now use a dummy value
   */
- ecmd->supported = SUPPORTED_10000baseT_Full;
+ ecmd->supported = 0;
  ecmd->autoneg = AUTONEG_DISABLE;
  ecmd->transceiver = XCVR_DUMMY1;
  ecmd->port = PORT_NONE;
-- 
1.7.1